### PR TITLE
Provide Kotlin DSL samples

### PR DESCRIPTION
**Pull request by eriwen**
Convert this as part of https://github.com/gradle/guides/issues/191

### CODE REVIEW

File: samples/groovy-dsl/code/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// tag::apply-plugin[]
+plugins {
+    id 'com.android.application'
+}
+// end::apply-plugin[]
+
+// tag::configure-plugin[]
+android {
+    compileSdkVersion 26
```
Comment (autonomousapps) [inline samples/groovy-dsl/code/app/build.gradle]: 28 is the latest stable

File: samples/groovy-dsl/code/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    defaultConfig {
+        applicationId "org.gradle.helloworldgradle"
+        minSdkVersion 21
```
Comment (autonomousapps) [inline samples/groovy-dsl/code/app/build.gradle]: is min 21 required? I expect a lot of Android apps are using minSdkVersion <19

File: samples/groovy-dsl/code/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+// end::configure-plugin[]
+
+// tag::dependencies[]
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation 'com.android.support:appcompat-v7:26.1.0'
```
Comment (autonomousapps) [inline samples/groovy-dsl/code/app/build.gradle]: if you bump the compile from 26 to 28, then this needs to be [28.0.0](https://developer.android.com/topic/libraries/support-library/revisions) (major version of support lib must match compile version).

File: samples/groovy-dsl/code/build.gradle [hunk 4]
```diff
@@ -0,0 +1,25 @@
+// Top-level build file where you can add configuration options common to all sub-projects/modules.
+
+buildscript {                 // <1>
+    repositories {
+        google()
+        jcenter()
+    }
+    dependencies {            // <2>
+        classpath 'com.android.tools.build:gradle:3.0.1'
```
Comment (autonomousapps) [inline samples/groovy-dsl/code/build.gradle]: latest version is 3.2.1. But it's probably ok to just use these default values. I assume this is from the new-project wizard in Android Studio 3.0.1?

File: samples/groovy-dsl/code/app/build.gradle (hunk 2 above)
Comment (eriwen) [reply samples/groovy-dsl/code/app/build.gradle]: My understanding is that this improves performance. Isn't that right?

File: samples/groovy-dsl/code/build.gradle (hunk 4 above)
Comment (eriwen) [reply samples/groovy-dsl/code/build.gradle]: I actually don't know. I didn't write the initial build I just moved this to a dedicated sample.

Comment (eriwen) [reply samples/groovy-dsl/code/build.gradle]: I actually don't know. I didn't write the initial build I just moved this to a dedicated sample.

File: samples/groovy-dsl/code/app/build.gradle (hunk 2 above)
Comment (autonomousapps) [reply samples/groovy-dsl/code/app/build.gradle]: Yes, I believe so. But it also has implications for accessibility. If your app has a minSdk of 21, then only Android _devices_ 21+ can install your app (meaning that, as a purveyor of fine apps, you're reducing your potential user base). 21 == Lollipop / Android 5.0. See [here](https://en.wikipedia.org/wiki/Android_version_history) for a reference document on this.

Comment (autonomousapps) [reply samples/groovy-dsl/code/app/build.gradle]: Bottom line: it's fine to do this. I was just asking of 21 was required. I wouldn't want to imply it was necessary if a lower number were acceptable.

**Pull request merged**